sdgym/datasets.py

```diff
@@ -17,21 +17,35 @@
 TIMESERIES_FIELDS = ['sequence_index', 'entity_columns', 'context_columns', 'deepecho_version']
 
 
-def download_dataset(dataset_name, datasets_path=None, bucket=None):
+def download_dataset(dataset_name, datasets_path=None, bucket=None, aws_key=None, aws_secret=None):
     datasets_path = datasets_path or DATASETS_PATH
     bucket = bucket or BUCKET
-    url = BUCKET_URL.format(bucket) + f'{dataset_name}.zip'
 
-    LOGGER.info('Downloading dataset %s from %s', dataset_name, url)
-    response = urllib.request.urlopen(url)
-    bytes_io = io.BytesIO(response.read())
+    LOGGER.info('Downloading dataset %s from %s', dataset_name, bucket)
+    s3 = _get_s3_client(aws_key, aws_secret)
+    obj = s3.get_object(Bucket=bucket, Key=f'{dataset_name}.zip')
+    bytes_io = io.BytesIO(obj['Body'].read())
 
     LOGGER.info('Extracting dataset into %s', datasets_path)
     with ZipFile(bytes_io) as zf:
         zf.extractall(datasets_path)
 
 
-def _get_dataset_path(dataset, datasets_path, bucket=None):
+def _get_s3_client(aws_key=None, aws_secret=None):
+    if aws_key is not None and aws_secret is not None:
+        # credentials available
+        return boto3.client(
+            's3',
+            aws_access_key_id=aws_key,
+            aws_secret_access_key=aws_secret
+        )
+    else:
+        # no credentials available, make unsigned requests
```

Comment:
The current proposal has the problem that if credentials are available system-wide (or as environment variables) but `aws_key` or `aws_secret` are not passed, an anonymous client will be created, preventing access to a private bucket that would otherwise be available.

I would add a new `if` here, inside the `else`, which checks for system-wide credentials before falling back to the unsigned client.

```python3
else:
    if boto3.Session().get_credentials():
        # credentials available and will be detected automatically
        config = None
    else:
        # no credentials available, make unsigned requests
        config = botocore.config.Config(signature_version=botocore.UNSIGNED)

    return boto3.client('s3', config=config)
```